dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/cdiproxies/CDIProxyServlet.java

```diff
@@ -0,0 +1,90 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package io.openliberty.microprofile.telemetry.internal_fat.apps.cdiproxies;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.util.logging.Logger;
+
+import org.junit.Test;
+
+import componenttest.app.FATServlet;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.Context;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.ClientBuilder;
+
+@WebServlet("/")
+@ApplicationScoped
+public class CDIProxyServlet extends FATServlet {
+
+    Logger logger = Logger.getLogger(this.getClass().getCanonicalName());
+
+    @Inject
+    Tracer tracer;
+
+    @Inject
+    Span injectedSpan;
+
+    @Inject
+    Baggage injectedBaggage;
+
+    @Test
+    public void testSpanUpdates() {
+        injectedSpan.toString(); //Just poke the object so CDI fetches the real object behind the proxy.
+
+        Span span = Span.current();
+        logger.info("Span.current at the start of hte method : " + span.toString());
+        assertEquals("The injected span was not the same as the current span before calling makeCurrent()", span.toString(), injectedSpan.toString()); //TODO - there's probably a better way to compare spans than toStrings (but just using (span, injectedSpan) will compare a weld proxy to a real Span object.
+
+        Span newSpan = tracer.spanBuilder("my span").setParent(Context.current().with(span)).startSpan();
```

Comment:
There's no need to set the parent here, it will automatically use the current span as the parent.

If you were to do it manually, I think you'd just want `Context.current()` here, but in this case you should just take it out.